pkg/service/service.go

```diff
@@ -603,7 +603,7 @@ func (s *Service) upsertServiceIntoLBMaps(svc *svcInfo, onlyLocalBackends bool,
 
 	err := s.lbmap.UpsertService(
 		uint16(svc.frontend.ID), svc.frontend.L3n4Addr.IP,
-		svc.frontend.L3n4Addr.L4Addr.Port,
+		svc.frontend.L3n4Addr.L4Addr.Port, string(svc.frontend.L3n4Addr.L4Addr.Protocol),
```

Comment:
There are some unit tests for this in `service_test.go`. I think they could be extended to add a new frontend with just the protocol changed.

Reply:
These look to be unit tests and the lbmap service is just the mock service, so I don't know how much that will really tell us. Is there something else that you're referencing?

Reply:
Ah, no you're right.

Reply:
I'd still add a unit test to check whether a svc doesn't get overwritten in pkg/service/service.go internal state if the same svc but with different proto gets added.